test/e2e/terminal.test.ts

```diff
@@ -0,0 +1,77 @@
+import { test, expect } from "@playwright/test"
+import * as fs from "fs"
+import { tmpdir } from "os"
+import * as path from "path"
+import util from "util"
+import * as cp from "child_process"
+import { STORAGE } from "../utils/constants"
+import { CodeServer } from "./models/CodeServer"
+
+test.describe("Integrated Terminal", () => {
+  // Create a new context with the saved storage state
+  // so we don't have to logged in
+  const options: any = {}
+  const testFileName = "test.txt"
+  const testString = "new string test from e2e test"
+  let codeServer: CodeServer
+  let tmpFolderPath: string = ""
+  let tmpFile: string = ""
+
+  // TODO@jsjoeio
+  // Fix this once https://github.com/microsoft/playwright-test/issues/240
+  // is fixed
+  if (STORAGE) {
+    const storageState = JSON.parse(STORAGE) || {}
+    options.contextOptions = {
+      storageState,
+    }
+  }
+  test.beforeEach(async ({ page }) => {
+    codeServer = new CodeServer(page)
+    await codeServer.setup()
+    // NOTE@jsjoeio
+    // We're not using tmpdir from src/node/constants
+    // because Playwright doesn't fully support ES modules from
+    // the erorrs I'm seeing
+    tmpFolderPath = fs.mkdtempSync(path.join(tmpdir(), "code-server-test"))
+    tmpFile = `${tmpFolderPath}${path.sep}${testFileName}`
+  })
+
+  test.afterEach(async () => {
+    // Ensure directory was removed
+    fs.rmdirSync(tmpFolderPath, { recursive: true })
```

Comment:
If we decide to make the path change above we''ll want to remove the directory for this test only:

```suggestion
    await fs.rmdir(testdir, { recursive: true })
```

Also, any thoughts on moving this to `beforeAll`? My thinking is that it could be useful for debugging in some scenarios to see the files generated by the tests.

Reply:
Good idea! Will move